server.cjs

```diff
@@ -0,0 +1,51 @@
+const { createServer, ServerResponse, IncomingMessage } = require('http')
+const { parse } = require('url')
+const next = require('next')
+ 
+const dev = process.env.NODE_ENV !== 'production'
+const hostname = 'localhost'
+const port = 3000
+// when using middleware `hostname` and `port` must be provided below
+const app = next({ dev, hostname, port })
+const handle = app.getRequestHandler()
+
+app.prepare().then(() => {
+  createServer(async (req, res) => {
+    // Be sure to pass `true` as the second argument to `url.parse`.
+    // This tells it to parse the query portion of the URL.
+    const parsedUrl = parse(req.url, true)
+
+    const setHeader = res.setHeader
+    let cacheControlOverwritten = false
+    res.setHeader = function(header, value) {
+      if (header.toLowerCase() === 'x-cache-control') {
+        cacheControlOverwritten = true
+        return setHeader.call(this, 'Cache-Control', value)
+      }
+
+      if (header.toLowerCase() === 'cache-control') {
+        if (cacheControlOverwritten) {
+          return this
+        }
+
+        // patch default next.js cache
+        const match = value.match(/s-maxage=(\d+), stale-while-revalidate/)
+        if (match) {
+          const maxAge = match[1]
+          return setHeader.call(this, header, `public, s-maxage=${maxAge}, stale-if-error=1200, stale-while-revalidate=30`)
```

Comment:
What should be the higher boundary? 1 day? 1 week? What if it matches any kind of static? I guess we want to keep static cached as long as we can 🤔

Reply:
I played a bit with this, it's not possible to catch these settings in dev mode to display alert in logs, since cache-control is specific for dev mode.

<img width="866" alt="image" src="https://github.com/lidofinance/trp-ui/assets/103929444/f942d40c-177a-422f-8a65-c7dd86f08312">

on the other hand, silently cutting maxage in production is a bit too late  🤔  Feels like this should be caught during tests or via linter 🤔